Core/TransactionHandler.php

```diff
@@ -0,0 +1,305 @@
+        ResponseHandler::onSuccessResponse($oResponse, $oBackendService, $oOrder);
+        $this->_updateParentTransactionStateIfNecessary($oResponse);
+        OrderHelper::updateOrderState($oOrder, $oResponse, $oBackendService);
+
+        return $this->_getSuccessMessage();
+    }
+
+    /**
+     * Handles the failure response case.
+     *
+     * @param FailureResponse $oResponse
+     *
+     * @return array
+     *
+     * @since 1.0.1
+     */
+    private function _onActionFailure($oResponse)
+    {
+        $sErrorDescriptions = $this->_getErrorString($oResponse);
+        return $this->_getErrorMessage(Helper::translate('text_generic_error') . '<br>' . $sErrorDescriptions);
+    }
+
+    /**
+     * Iterates over all errors in the FailureResponse and returns the descriptions as a string.
+     *
+     * @param FailureResponse $oResponse
+     *
+     * @return string
+     *
+     * @since 1.0.1
+     */
+    private function _getErrorString($oResponse)
+    {
+        $sErrors = '';
+
+        foreach ($oResponse->getStatusCollection()->getIterator() as $oItem) {
+            $sErrors .= $oItem->getDescription() . "<br>";
+            $this->_oLogger->error($oItem->getCode() . ': ' . $oItem->getDescription());
+        }
+
+        return $sErrors;
+    }
+
+    /**
+     * Checks if the parent transaction still has money to be captured.
+     * If no money is available anymore, the state is set to closed.
+     *
+     * @param SuccessResponse $oResponse
+     *
+     * @since 1.0.1
+     */
+    private function _updateParentTransactionStateIfNecessary($oResponse)
+    {
+        $sParentTransactionId = $oResponse->getParentTransactionId();
+
+        $oParentTansaction = oxNew(Transaction::class);
+        $oParentTansaction->loadWithTransactionId($sParentTransactionId);
+
+        $fRestAmount = $this->getTransactionMaxAmount($sParentTransactionId);
+
+        if ($fRestAmount <= 0) {
+            $oParentTansaction->wdoxidee_ordertransactions__state = new Field(Transaction::STATE_CLOSED);
+            $oParentTansaction->save();
+        }
+    }
+
+    /**
+     * Calculates the maximum amount available for a transaction.
+     *
+     * @param string $sTransactionId
+     *
+     * @return float
+     *
+     * @since 1.0.1
+     */
+    public function getTransactionMaxAmount($sTransactionId)
+    {
+        $oTransaction = oxNew(Transaction::class);
+        $oTransaction->loadWithTransactionId($sTransactionId);
+        $fBaseAmount = $oTransaction->wdoxidee_ordertransactions__amount->value;
+
+        // get the child transactions amount summed
+        $oDb = DatabaseProvider::getDb();
+        $oDb->setFetchMode(DatabaseProvider::FETCH_MODE_ASSOC);
+
+        $sDbIdentifier = $oDb->quoteIdentifier('wdoxidee_ordertransactions');
+
+        $sDbQuery = "SELECT SUM(amount) AS childTransactionsTotalAmount FROM {$sDbIdentifier}
+                        WHERE PARENTTRANSACTIONID LIKE ?";
```

Comment:
why `LIKE`? shouldn't it be a `=`. We need a strong match, right?

Reply:
Excellent point, thanks 👍